app/api/user.py

```diff
@@ -43,3 +43,101 @@ def user_assign_role(user_id, role):
+            user_ids = []
+            for user in users:
+                user_ids.append(str(user['_id']))
+            # Now find weekly reports of all those users_id in the above list whose report is not reviewd   
+            for data in  user_ids:
+                docs = mongo.db.reports.find({
+                    "user":str(data),
+                    "review": {'$exists': False},
+                    "type" : "weekly",
+                    "created_at": {
+                    "$gte": datetime.datetime(last_monday.year, last_monday.month, last_monday.day),
+                    "$lte": datetime.datetime(last_sunday.year, last_sunday.month, last_sunday.day)
+                     }  
+                })
+                docs = [serialize_doc(doc) for doc in docs]
+                # Append those user records whose not been reviewd in user_id list
+                user_id = []
+                for user in docs:
+                    user_id.append(user)
+                # Then if we find that data exist in user_id report than update the manager recent activity with a date and message    
+                if user_id is not None: 
+                    ret = mongo.db.recent_activity.update({
+                    "user": str(current_user["_id"]),},
+                    {"$set":{"recent_activity":{
+                    "created_at": datetime.datetime.now(),
+                    "Message":"Your have not reviewed your juniors weekly report"
+                    }}},upsert=True)
+                return jsonify(str(ret))       
+        # Here we see the recent activity of normal employee or whose role is not manager or admin    
+        else:
+            # First let's find the weekly reports who have been reviewd by manager 
+            docs = mongo.db.reports.find({
+                "user": str(current_user["_id"]),
+                "review": {'$exists': True},
+                "type" : "weekly",
+                "created_at": {
+                "$gte": datetime.datetime(last_monday.year, last_monday.month, last_monday.day),
+                "$lte": datetime.datetime(last_sunday.year, last_sunday.month, last_sunday.day)
+            }
+            })
+            # Here see if we find a reviewed report in  a particular time from the above condition 
+            docs = [serialize_doc(doc) for doc in docs]
+            if not docs:
+                return("review not available")
+            # if report is find reviewd which are generated within a week set a recent activity for normal employee with datetime and message
+            else:
+                ret = mongo.db.recent_activity.update({
+                    "user": str(current_user["_id"]),},
+                    {"$set":{"recent_activity":{
+                    "created_at": datetime.datetime.now(),
+                    "user": str(current_user["_id"] ),
+                    "Message":"Your weekly report has been reviewed"

```

Comment:
there would be multiple managers for an employee. so need to show exactly. who has reviewed weekly reports